content/blog/tidymodels-july-2021/index.Rmd

```diff
@@ -0,0 +1,134 @@
+
+photo:
+  url: https://unsplash.com/photos/AChwtt3tBPU
+  author: Danil Shostak
+
+# one of: "deep-dive", "learn", "package", "programming", or "other"
+categories: [package] 
+tags: [parsnip, tune, workflows, hardhat, discrim, finetune, workflowsets]
+---
+
+<!--
+TODO:
+* [ ] Look over / edit the post's title in the yaml
+* [ ] Edit (or delete) the description; note this appears in the Twitter card
+* [ ] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [ ] Find photo & update yaml metadata
+* [ ] Create `thumbnail-sq.jpg`; height and width should be equal
+* [ ] Create `thumbnail-wd.jpg`; width should be >5x height
+* [ ] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+Recently, we had a series of CRAN releases: [hardhat](https://hardhat.tidymodels.org/news/index.html#hardhat-0-1-6-2021-07-14), [workflows](https://workflows.tidymodels.org/news/#workflows-0-2-3-2021-07-15), [parsnip](https://parsnip.tidymodels.org/news/#parsnip-0-1-7-2021-07-21), [tune](https://tune.tidymodels.org/news/#tune-0-1-6-2021-07-21), [finetune](https://finetune.tidymodels.org/news/#finetune-0-1-0-unreleased), [workflowsets](https://workflowsets.tidymodels.org/news/#workflowsets-0-1-0-unreleased), and [discrim](https://discrim.tidymodels.org/news/#discrim-0-1-3-unreleased). These were coordinated because of some cross-package improvements. This blog post summarizes the changes. 
+
+## Object extraction
+
+The tidymodels team decided that we needed a consistent set of APIs for extracting things from objects. For example, a parsnip model has the underlying model fit based on the engine. For example, `linear_reg()` with the `"lm"` engine contains an `lm` object. There were some existing functions to do this (mostly named `pull_*()`) but they were fairly inconsistent and were not generics. 
+
+We added the following functions: `extract_fit_engine()`, `extract_fit_parsnip()`, `extract_mold()`, `extract_numeric()`, `extract_preprocessor()`, `extract_recipe()`, `extract_spec_parsnip()`, `extract_workflow()`, and `extract_workflow_set_result()`. 
+
+The nice thing about this change is that a function such as `extract_recipe()` can be used with objects created by the tune, workflows, or workflowsets packages. 
+
+The existing `pull_*()` methods have been soft-deprecated and will stick around for a while. 
+
+## Better model documentation
+
+One issue that we've seen in the parsnip documentation is that there is just _so much_ on each model page. It can be intimidating and difficult to find that one piece of information that you were looking for. 
+
+We've reorganized the model pages so that there are now sub-packages for each engine. For example, when you use `?linear_reg`, the help page has a _dynamic_ list of engines from parsnip or any parsnip-adjacent package that has been loaded. Here is what the pkgdown site looks like: 
```

Comment:
```suggestion
We've reorganized the model pages so that there are now sub-packages for each engine. For example, when you use `?linear_reg`, the help page has a _dynamic_ list of engines from parsnip or any parsnip-adjacent package that has been loaded. Here is what the [pkgdown site](https://parsnip.tidymodels.org/reference/linear_reg.html) looks like: 
```

Might be nice to link directly to the site from here